pygeo/parameterization/DVGeoCST.py

```diff
@@ -0,0 +1,1093 @@
+
+
+class DVGeometryCST(BaseDVGeometry):
+    r"""
+    This class implements a 2D geometry parameterisation based on Brenda Kulfan's CST (Class-Shape Transformation) method.
+    This class can work with 3D coordinates but will only change the point coordinates in one direction.
+
+    The CST equation is as follows:
+
+    :math:`y(x) = C(x) * S(x) + y_\text{te}x`
+
+    Where C is the class function:
+
+    :math:`C(x) = (x^{N1} + (1 - x)^{N2})`
+
+    And S is the shape function, in this case a summation of Bernstein polynomials:
+
+    :math:`S(x) = \sum_i^n w_i \binom{n}{i}x^i(1-x)^{n-i}`
+
+    Here x is the normalized chordwise coordinate, ranging from 0 to 1 from front to the rear of the shape.
+
+    This class works for extruded 2D airfoil geometries, but not 3D geometries that vary in the spanwise direction.
+
+    Parameters
+    ----------
+    datFile : str
+        Filename of dat file that represents the initial airfoil. The coordinates in this file will be used to
+        determine the camber line, which is the dividing line to distinguish upper and lower surface points.
+    idxChord : int, optional
+        Index of the column in the point set to use as the chordwise (x in CST) coordinates, by default 0
+    idxVertical : int, optional
+        Index of the column in the point set to use as the vertical (y in CST) airfoil coordinates, by default 1
+    comm : MPI communicator, optional
+        Communicator for DVGeometryCST instance, by default MPI.COMM_WORLD
+    isComplex : bool, optional
+        Initialize variables to complex types where necessary, by default False
+    debug : bool, optional
+        Show plots when addPointSet is called to visually verify that it is correctly splitting
+        the upper and lower surfaces of the airfoil points, by default False
+    """
+
+    def __init__(self, datFile, idxChord=0, idxVertical=1, comm=MPI.COMM_WORLD, isComplex=False, debug=False):
+        super().__init__(datFile)
+        self.xIdx = idxChord
+        self.yIdx = idxVertical
+        self.comm = comm
+        self.isComplex = isComplex
+        if isComplex:
+            self.dtype = complex
+            self.dtypeMPI = MPI.DOUBLE_COMPLEX
+        else:
+            self.dtype = float
+            self.dtypeMPI = MPI.DOUBLE
+        self.debug = debug
+        if debug and not pltImport:
+            raise ImportError("matplotlib.pyplot could not be imported and is required for DVGeoCST debug mode")
+
+        # Store the DVs and flags to determine if the limited options have already been specified
+        self.DVs = {}
+        self.DVExists = {
+            "upper": False,
+            "lower": False,
+            "n1_upper": False,
+            "n2_upper": False,
+            "n1_lower": False,
+            "n2_lower": False,
+            "chord": False,
+        }
+
+        # Default DVs to be copied for each point set
+        self.defaultDV = {
+            "upper": 0.1 * np.ones(8, dtype=self.dtype),
```

Comment:
Is this assuming we will always have 8 coefficients to control? If so, I think this should be an input argument at least. I am also confused on how this interacts with the DV addition. Is the user expected to stay below 8 per half airfoil?